File: DAY-5/isSorted_recursion.cpp
#include<bits/stdc++.h>
using namespace std;

bool isSorted(int arr[],int index,int n){
	if(index==n-1){
		return true;
	}

	return arr[index]<arr[index+1] && isSorted(arr,++index,n);
}

int main(){
	int arr[]={1,2,23,42,67,89};
	int n=sizeof(arr)/sizeof(arr[0]);
	cout<<(isSorted(arr,0,n)?"Yes":"No");	
}

File: DAY-5/isSorted_recursion.java
package com.CherishDSA;

public class isSorted_recursion {
    public static boolean isSorted(int[] arr,int index,int n){
        if(index==n-1){
            return true;
        }
        return arr[index]<arr[index+1] && isSorted(arr,++index,n);
    }
    public static void main(String[] args) {
        int[] arr={2,34,56,78,1829,400};
        System.out.println(isSorted(arr,0,arr.length));
    }
}
